### DISCUSSION TITLE
iOS device location does not work on TestFlight deployments made via Codemagic, but do on TestFlight deployments made via local Xcode

### DISCUSSION THREAD

**Discussion by nlistgarten** [codemagic-ci-cd/codemagic-docs#1926] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am writing a Flutter app for iOS and Android with location information in it. The location information no longer works properly on TestFlight builds built and submitted through CodeMagic. It used to work fine, but now it doesn't, and I can't figure out why. The only significant change I made in the codebase was upgrading the Koltin version from 1.7 to 1.9 in the Android build.gradle, which should not affect the iOS build at all. Building and submitting the exact same bundle from Xcode locally, then installing from TestFlight works fine. I am _guessing_ that the issue is related to signing or similar in my Codemagic set up, but am not sure how to troubleshoot it.
What doesn't work:
- Build + distribute to TestFlight via Codemagic
What does work:
- Build + distribute to TestFlight via my local Xcode
Both Codemagic's build configuration and my local Xcode are Version 15.0 beta 4 (15A5195m). How do I go about troubleshooting/identifying the root cause of this issue or figuring out a workaround?

### Expected results

Installed app will use device's location no matter if that app is built on Xcode on my local machine or via Codemagic.

### Actual results

Installed app only uses device's location if the build is made on Xcode on my local machine, not via Codemagic.

### Build id (optional)

Issue began occurring around build 64bdfd13f76100fe5c00c96b or 64bdff210d5d2d5797fc8de2

**Comment by icarusdust** [comment]
Hi @nlistgarten, I'd need you to elaborate on what you exactly mean by it is not working. What exactly happens after installing a binary built through Codemagic ?

**Reply by nlistgarten** [reply]
I have some code that attempts to get a location, but uses a default value if no location is found (whether that's permissions being denied or inability for the phone to get its location). I am using both the flutter packages "location" and "geolocator" (I have a goal to move entirely to geolocator but I haven't done that yet, but that's neither here nor there).
- With a local Xcode build uploaded directly to TestFlight: both location packages accurately return a location, which is used in the app
- With a Codemagic build uploaded within Codemagic to TestFlight: Neither location package returns a valid location, and the default "no location" information is used.
- With a Codemagic build uploaded via Transporter directly to TestFlight: Neither location package returns a valid location, and the default "no location" information is used.

The fact that two packages are affected makes me think it may be a signing issue. Not quite sure why that would be or how to fix it, though

**Comment by nlistgarten** [comment]
Solved: the reason that this did not work was tied to an issue initializing the app. That issue was caused by a different package entirely (connectivity plus) due to a [known issue](https://github.com/fluttercommunity/plus_plugins/issues/1955) with Xcode 15 beta 4 on devices iOS 16 and earlier.
Following the workaround steps listed and dropping the Xcode build version back to 14.3.1 resolved the issue.
